src/main/java/mate/academy/onlinebookstore/service/BookServiceImpl.java

```diff
@@ -0,0 +1,38 @@
+package mate.academy.onlinebookstore.service;
+
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import mate.academy.onlinebookstore.dto.BookDto;
+import mate.academy.onlinebookstore.dto.CreateBookRequestDto;
+import mate.academy.onlinebookstore.exception.DataProccesingException;
+import mate.academy.onlinebookstore.mapper.BookMapper;
+import mate.academy.onlinebookstore.model.Book;
+import mate.academy.onlinebookstore.repository.BookRepository;
+import org.springframework.stereotype.Service;
+
+@RequiredArgsConstructor
+@Service
+public class BookServiceImpl implements BookService {
+    private final BookRepository bookRepository;
+    private final BookMapper bookMapper;
+
+    @Override
+    public BookDto save(CreateBookRequestDto requestDto) {
+        Book book = bookMapper.toModel(requestDto);
+        return bookMapper.toDto(bookRepository.save(book));
+    }
+
+    @Override
+    public List<BookDto> findAll() {
+        return bookRepository.findAll().stream()
+                .map(bookMapper::toDto)
+                .toList();
+    }
+
+    @Override
+    public BookDto getBookById(Long id) {
+        Book book = bookRepository.findById(id).orElseThrow(
+                () -> new DataProccesingException("Can't not find book by id " + id));
```

Comment:
EntityNotFoundException for service layer

```suggestion
                () -> new EntityNotFoundException("Can't not find book by id " + id));
```